src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/FindPaginationTableIntegrationTest.java

```diff
@@ -30,12 +39,134 @@ public final void dropScenario() {
+    // The return data has been verified in the previous test, so this test only checks the page
+    // state
+    String nextPage =
+        assertTableCommand(keyspaceName, TABLE_NAME)
+            .templated()
+            .find(Map.of(), List.of(), Map.of())
+            .wasSuccessful()
+            .hasNextPageState()
+            .extractNextPageState();
 
+    // It works well with the next page state
+    Map<String, Object> options = ImmutableMap.of("pageState", nextPage);
     assertTableCommand(keyspaceName, TABLE_NAME)
         .templated()
-        .find(Map.of(), List.of())
+        .find(Map.of(), List.of(), Map.of(), options)
         .wasSuccessful()
         .hasNextPageState();
   }
+
+  @Test
+  @Order(3)
+  public void findWithCQLSortAndPageState() {
+    // find command with cql sort and extract page state
+    Map<String, Object> sort =
+        ImmutableMap.of(fieldName(PartitionedKeyValueTable50Scenario.CLUSTER_COL_1), -1);
+
+    String nextPage =
+        assertTableCommand(keyspaceName, TABLE_NAME)
+            .templated()
+            .find(FILTER_ID, List.of(), sort)
+            .wasSuccessful()
+            .hasNextPageState()
+            .extractNextPageState();
+
+    // use the page state from the above without any error
+    Map<String, Object> options = ImmutableMap.of("pageState", nextPage);
+    assertTableCommand(keyspaceName, TABLE_NAME)
+        .templated()
+        .find(FILTER_ID, List.of(), sort, options)
+        .wasSuccessful()
+        .doesNotHaveNextPageState();
+
+    // ok to pass empty page state which will be ignored, and return the same page state
+    options = ImmutableMap.of("pageState", "");
+    String pageState =
+        assertTableCommand(keyspaceName, TABLE_NAME)
+            .templated()
+            .find(FILTER_ID, List.of(), sort, options)
+            .wasSuccessful()
+            .hasNextPageState()
+            .extractNextPageState();
+    Assertions.assertEquals(
+        nextPage,
+        pageState,
+        "The page state should match when empty string is passed as page state");
+  }
+
+  @Test
+  @Order(4)
+  public void findWithInMemorySortAndPageState() {
+    // find command with in memory sort and it should not have page state
+    Map<String, Object> sort =
+        ImmutableMap.of(fieldName(PartitionedKeyValueTable50Scenario.VALUE_COL), 1);
+
+    // should not have page state returned
+    assertTableCommand(keyspaceName, TABLE_NAME)
+        .templated()
+        .find(FILTER_ID, List.of(), sort)
+        .wasSuccessful()
+        .doesNotHaveNextPageState();
+
+    // error if page state is passed with in memory sort
+    Map<String, Object> options = ImmutableMap.of("pageState", "AAgABAAAAOYBAYDn8H///+s=");
+    assertTableCommand(keyspaceName, TABLE_NAME)
+        .templated()
+        .find(FILTER_ID, List.of(), sort, options)
+        .hasSingleApiError(
+            SortException.Code.UNSUPPORTED_PAGINATION_WITH_IN_MEMORY_SORTING,
```

Comment:
a better test would check that the message is specific to the request. 

We keep lines in the error message like this one: 
`The command sorted on the columns: ${sortColumns}.`

for that, so check the line has the name of the sort column(s)

Reply:
prob also need a multi column sort